### DISCUSSION TITLE
How to equally divide the images in selection at offset X,Y and width/height?

### DISCUSSION THREAD

**Discussion by snowman** [ImageMagick/ImageMagick#5992] [Help] [answered]
### ImageMagick version

7.1.0-47

### Operating system, version and so on

Microsoft Windows 10 Version 21H2 (OS Build 19044.1826)

### Description

There is a file `example.jpg` that has 13x13 dimensions.

How to equally divide/crop `example.jpg` with the selection at X: 2, Y:4 (0-based offset), and selection size width and height: 9x2 into 3x2 tiles, each tile has 3x1 dimensions?

My try has failed:
```sh
$ magick example.jpg -repage 9x2-2-4 -crop 3x2@ tile_%d.jpg
```

![image](https://user-images.githubusercontent.com/16096559/213867330-1f6f911f-48ec-4155-8922-1899a4860e6b.png)

**Comment by snibgo** [comment, marked as answer]
I suggest cropping twice. The first creates the 9x2 image, and the second is as you have it, cropping into 6 pieces.
```
magick example.jpg -crop 9x2+2+4 +repage -crop 3x2@ +repage tile_%d.jpg
```

**Reply by snowman** [reply]
simple and well done :)

**Comment by GeeMack** [comment]
This command will crop an input image into 13 rows and 13 columns, 169 pieces. Then it deletes all except the ones you've highlighted with color in your example. After that it appends those remaining into a single horizontal row, and finishes by cropping that into the 6 colored segments shown in your input image.

    magick input.png -crop 13x13@ -delete 0-53,63-66,76--1 +append -crop 6x1@ +repage tile%02d.png



This command starts by shaving 4/13 of the height from both the top and bottom, and 2/13 of the width from each side to isolate just the working area (plus a few more units...). Then it crops that result into 3x1 unit pieces. Then it deletes the unneeded pieces and keeps the 6 colored segments shown in your sample image.

    magick input.png -shave %[fx:w*2/13]x%[fx:h*4/13] -crop 3x5@ -delete 6--1 +repage tile%02d.png

The output images of these commands will be named tile01.png, tile02.png... etc.

Keep in mind when cropping into a certain number of units with  the "@" flag, there may be a single pixel difference in the dimensions of the output images if the input isn't evenly divisible by the number of units.
